core/src/blocktree_processor.rs

```diff
@@ -87,14 +89,15 @@ pub fn process_entries(bank: &Bank, entries: &[Entry]) -> Result<()> {
         // else loop on processing the entry
         loop {
             // try to lock the accounts
-            let lock_results = bank.lock_accounts(&entry.transactions);
+            let txs_execution_order = generate_random_order(entry.transactions.len());
```

Comment:
This function replays transactions on the validator side, which means it must deterministically lock/play entries, otherwise the validators will diverge from each other.

Reply:
@carllin once they are locked, they can be executed in random order.  So the memory is unlikely to be placed in the same spot on all the validators.

Reply:
@aeyakovenko, ah ok, I assumed maybe that was the implication. Right now after we lock, the entries are handed off to a par_iter(), so it is sort of random the order in which order they execute.

Reply:
@carllin I think its still sequential within a batch.

Reply:
@carlin entries are being processed in parallel, but txs containing in entry object are processed in batch. My changes radomize that.

Reply:
@ParthDesai, makes sense now, thanks!